Assets/Scripts/Unit/Action/Move/Frames/MoveFrame.cs

```diff
@@ -3,31 +3,47 @@
 using System.Collections.Generic;
 
 public class MoveFrame : Frame {
-	public override bool Execute(Unit unit, Direction dir, Board board) {
-		Direction absoluteDir = (Direction)(((int)relativeDir + (int)dir) % 4);
-		Vector2 next = nextCoord(unit.tile.coordinate, absoluteDir);
+	public override bool CanExecute(Unit unit, Direction dir, Board board) {
+		Vector2 next = GetRelativeDisplacement(unit, dir, board);
+		return board.CheckCoord(next);
```

Comment:
This checks if the relative displacement is a valid coord when it seems like it should be checking the absolute displacement

Reply:
In this case there should be no difference between RelativeDisplacement and AbsoluteDisplacement.

Reply:
Scratch that, I mixed up the terminology. What I meant to say is that CheckCoord is checking RelativeDisplacements but it should be checking the resultant destinations. e.g. if the unit was trying to go up, next would be (-1, 0) which would always fail CheckCoord. I think what you want to pass to CheckCoord is something like unit.tile.coordinate + next

Reply:
Oh you are right - yeah this is only checking the relative displacement but not the actual coordinate. I'll make that fix.